src/transformers/modeling_tf_utils.py

```diff
@@ -647,7 +647,7 @@ def strip_model_name_and_prefix(name, _prefix=None):
     return name
 
 
-def tf_shard_checkpoint(weights, max_shard_size="10GB"):
+def tf_shard_checkpoint(weights, max_shard_size="10GB", weights_name: str = TF2_WEIGHTS_NAME):
```

Comment:
Should we have the default shard size match the one in load_tf_weights?

Reply:
I got here by following the torch code, which also has the same issue! Specifically, `save_pretrained()` has a default size of 5GB, but the actual checkpoint sharding methods have a default size of 10GB. In general, though, the value passed from `save_pretrained()` will override those values.

It's a very minor detail either way, since I think both 5GB and 10GB shards work fine! We could consider standardizing everything at some point, but I don't think it's a high priority.